gradle/java.gradle

```diff
@@ -99,6 +99,12 @@ repositories {
 }
 
 dependencies {
+  testImplementation enforcedPlatform(group: 'org.junit', name: 'junit-bom', version: versions.junit5)
+  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api'
+  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-params'
+  testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine'
+  testRuntimeOnly group: 'org.junit.vintage', name: 'junit-vintage-engine'
```

Comment:
Do we need both old and new engines?

Reply:
We need new engine for any Java tests, we have one `PatchLoggerTest`. If we get to Spock 2.0 we'll remove vintage at that point.